src/persist/src/indexed/encoding.rs

```diff
@@ -33,10 +33,8 @@ pub struct Id(pub u64);
 /// - The updates field is non-empty.
 #[derive(Debug, Abomonation)]
 pub struct BufferEntry<K, V> {
-    /// Id of the stream this batch belongs to.
-    pub id: Id,
-    /// The updates themselves.
-    pub updates: Vec<((K, V), u64, isize)>,
+    /// Pairs of stream id and the updates themselves.
+    pub updates: Vec<(Id, Vec<((K, V), u64, isize)>)>,
```

Comment:
does it make sense to add an invariant that this is a set of ids, or do we need the generality of multiple instances of the same id?

Reply:
Neither, really. We don't need the generality but the invariant doesn't get us anything at the moment, either. I've add a note here and one at the user facing api.